### PR TITLE
Update constraints from upstream

### PR DESCRIPTION
**Pull request by xylar**
A constraint got missed on sqlalchemy and it's giving me trouble
with pip check downstream.

Add xylar as a maintainer

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by xylar** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml [hunk 1]
```diff
@@ -21,8 +21,8 @@ requirements:
     - setuptools
   run:
     - python >=3.6
-    - snowflake-connector-python <3
-    - sqlalchemy <2
+    - snowflake-connector-python <3.0.0
+    - sqlalchemy >=1.4.0,<2.0.0
```
Comment (xylar) [inline recipe/meta.yaml:25]: This bound got missed in #7 and is causing me trouble with pip check in https://github.com/conda-forge/apache-airflow-providers-snowflake-feedstock/pull/4

I'm a bit worried that the solver will just use build 0 when build 1 doesn't work.  But this is a pretty small thing to report build 0 as broken for.  Let's see how it goes.

**Comment by xylar** [comment]
@xhochy, could you take a look at this when you get a chance?  Thanks!

File: recipe/meta.yaml (hunk 1 above)
Comment (xhochy) [reply recipe/meta.yaml:25]: We should write a small repodata patch that fixes this. Then build 0 would also be usable.

**Review by xhochy — APPROVED** [review]

**Pull request merged**